doc/user/content/overview/architecture.md

```diff
@@ -1,31 +1,99 @@
+
+### Clusters
+
+Clusters are logical components that enable users to express resource isolation for objects responsible for computing and maintaining views (indexes and materialized views). A cluster specifies which indexes and materialized views share the same computational resources (CPU and RAM).
+
+However, a cluster is a logical respresentation of this relationship. Materialize will not actually spin up any computational resources when a cluster is created. Materialize will do so when a cluster is endowed with cluster replicas. A cluster can be active-active replicated by endowing it with multiple replicas. As long as one of the replicas is making progress, the cluster is making progress and can service reads and writes.
+
+### Cluster replicas
+
+Cluster replicas are the physical incarnations of a given cluster. A cluster replica represents a physical set of computational resources (CPU and RAM) that is responsible for maintaining the indexes and materialized views defined on the parent cluster.
+
+Each cluster replica associated with a given cluster contains all of the indexes and materialized views associated with that cluster. However, different cluster replicas can be differently sized, even if they are associated with the same cluster. Each replica is guaranteed to produce the same outputs as all of its peers for all times, because all cluster replicas read the same input data collections from the storage layer, and they all perform the same deterministic computations.
+
+### Indexes
+
+Indexes assemble and maintain source, view or materialized view output data in-memory on all cluster replicas associated with a given cluster. Creating an index enables future `SELECT` queries against the given source, view or materialized view to return rapidly because the requested data can be read without interacting with persistent storage, or incurring (much) additional computation.
+
+Users can create indexes on any source, unmaterialized view, or materialized view present in the region. However indexes are associated with a particular cluster, and when an index is created, altered, or destroyed, the corresponding change is replicated on all current and future replicas associated with that cluster. This setup lets users decouple changes to the computation (which views are defined, which indexes are created) from how that computation is physically deployed (how many replicas, how large is each replica).
+
+Note however, that unlike (materialized) views, indexes are local to a cluster and can can only be used on `SELECT` queries on the same cluster.
+
+For a deeper dive into how indexes work see [here](/ops/optimization/#indexes) and [here](/sql/create-index).
+
+### Materialized views
+
+Materialized views compute and then maintain views as data collections stored in persistent storage (S3). Subsequent `SELECT` queries can then read results from the stored data collection without having to perform (much) additional computation.
```

Comment:
Not clear what you mean by `(much)` here. AFAIK the amount of computation done is zero, except for Persist stuff (decoding the batches, consolidating if necessary, etc.)

Do you mean we might do additional computation if the `SELECT` on top of the materialized view is non-trivial? If so, that's true, but it doesn't really have to do with reading from the MV, and in fact it can be arbitrarily complicated.